newrelic/data_source_newrelic_account.go

```diff
@@ -0,0 +1,111 @@
+package newrelic
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/newrelic/newrelic-client-go/pkg/accounts"
+)
+
+func dataSourceNewRelicAccount() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNewRelicAccountRead,
+		Schema: map[string]*schema.Schema{
+			"scope": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      string(accounts.RegionScopeTypes.IN_REGION),
+				Description:  `The scope of the account in New Relic.  Valid values are "global" and "in_region".  Defaults to "in_region".`,
+				ValidateFunc: validation.StringInSlice([]string{"global", "in_region"}, true),
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of the account in New Relic.",
+			},
+			"account_id": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The ID of the account in New Relic.",
+			},
+		},
+	}
+}
+
+func dataSourceNewRelicAccountRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderConfig).NewClient
+
+	log.Printf("[INFO] Reading New Relic accounts")
+
+	scope := accounts.RegionScope(d.Get("scope").(string))
```

Comment:
Just a followup regarding the casing questions above. Do we need to handle casing here?